packages/client/components/Icon.tsx

```diff
@@ -1,28 +1,9 @@
 import styled from '@emotion/styled'
-import {ICON_SIZE} from '../styles/typographyV2'
+import {iconStyles} from '~/styles/helpers/iconStyles'
 
-const Icon = styled('i')({
-  flexShrink: 0,
-  fontFamily: 'Material Icons',
-  fontWeight: 'normal',
-  fontStyle: 'normal',
-  // Preferred icon size (24px)
-  fontSize: ICON_SIZE.MD24,
-  display: 'inline-block',
-  lineHeight: 1,
-  textTransform: 'none',
-  letterSpacing: 'normal',
-  wordWrap: 'normal',
-  whiteSpace: 'nowrap',
-  direction: 'ltr',
-  // Support for all WebKit browsers
-  WebkitFontSmoothing: 'antialiased',
-  // Support for Safari and Chrome
-  textRendering: 'optimizeLegibility',
-  // Support for Firefox
-  mozOsxFontSmoothing: 'grayscale',
-  // Support for IE
-  fontFeatureSettings: 'liga'
-})
+const Icon = styled.i`
+  ${iconStyles};
```

Comment:
is there a way to keep typesafety if we use this new pattern?

for example: 

```ts
const Icon = style('i')({
 textRendering: 'optimizeLegibility'
})
```
vscode will offer typeahead support for both the key & value & if you type an invalid value it'll be a typescript error.

alternatively, you could always subclass the component instead of the styles:
```ts
const IconOutlined = styled(Icon)({
  fontFamily: 'Material Icons Outlined'
})
```

Reply:
Valid concern! 
When using `styled.i` we're still getting eslint errors. 
![Screenshot 2021-08-31 at 15 08 25](https://user-images.githubusercontent.com/1017620/131516008-2092503f-9416-476f-a76c-aab8cb5dc7d6.png)
My idea was to prevent usage of unstyled icons and use the composition: https://emotion.sh/docs/composition

Right now I think it's better to keep things as they're and subclass the icons, ie. have a base `Icon` and then `IconFilled` and `IconOutlined`. Will push changes soon.

Reply:
@mattkrick I've just changed icons implementation to the way you mentioned. `IconOutlined` is simply created by extending & overriding `Icon`. I really overcomplicated it before 😓 . Thank you for the suggestion!